packages/cart-sdk/hooks/use-cart.ts

```diff
@@ -35,6 +36,26 @@ export function useCart() {
    return query;
 }
 
+export function extractCrossSells(cart: Cart | null | undefined) {
+   return useMemo(
```

Comment:
I noticed that this function contains a hook, therefore [by React convetions it should be a hook itselves](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions).

I'd take this opportunity to suggest to refactor like this (taking advantage of context, so where we use this we don't have to use another hook to get the cart):

```suggestion
export function useCartCrossSells() {
   const cart = useCart();
   return useMemo(
```

Reply:
Ok. I'm always a little leery about the performance effects of adding additional calls to contexts when the values we wan't already in scope. That said, I appreciate the encapsulation.

I'll go for it.